src/ProjectOrigin.Electricity/CommandOrchestrator.cs

```diff
@@ -0,0 +1,108 @@
+        return ToCommandStatus(command, result);
+    }
+
+    public async Task<Register.V1.CommandStatus> Process(Command<TransferProductionSliceCommand> command)
+    {
+        var step = new CommandStep<TransferProductionSliceCommand.Types.ProductionSliceTransferredEvent>(
+            command.Content.Event.CertificateId.ToModel(),
+            new SignedEvent<TransferProductionSliceCommand.Types.ProductionSliceTransferredEvent>(command.Content.Event, command.Content.Signature.ToByteArray()),
+            typeof(ProductionCertificate),
+            command.Content.Proof);
+
+        var result = await _stepProcessor.Process(step);
+
+        return ToCommandStatus(command, result);
+    }
+
+    public async Task<Register.V1.CommandStatus> Process(Command<ClaimCommand> command)
+    {
+
+        var result = await _stepProcessor.Process(
+            new CommandStep<ClaimCommand.Types.AllocatedEvent>(
+                command.Content.ProductionAllocatedEvent.ProductionCertificateId.ToModel(),
+                new SignedEvent<ClaimCommand.Types.AllocatedEvent>(command.Content.ProductionAllocatedEvent, command.Content.ProductionAllocatedSignature.ToByteArray()),
+                typeof(ProductionCertificate),
+                command.Content.ProductionAllocatedProof));
+
+        if (result.State != CommandStepState.Completed)
+            return ToCommandStatus(command, result);
+
+        result = await _stepProcessor.Process(
+            new CommandStep<ClaimCommand.Types.AllocatedEvent>(
+                command.Content.ConsumptionAllocatedEvent.ConsumptionCertificateId.ToModel(),
+                new SignedEvent<ClaimCommand.Types.AllocatedEvent>(command.Content.ConsumptionAllocatedEvent, command.Content.ConsumptionAllocatedSignature.ToByteArray()),
+                typeof(ConsumptionCertificate),
+                command.Content.ConsumptionAllocatedProof));
+
+        if (result.State != CommandStepState.Completed)
+            return ToCommandStatus(command, result);
+
+        result = await _stepProcessor.Process(
+            new CommandStep<ClaimCommand.Types.ClaimedEvent>(
+                command.Content.ProductionClaimedEvent.CertificateId.ToModel(),
+                new SignedEvent<ClaimCommand.Types.ClaimedEvent>(command.Content.ProductionClaimedEvent, command.Content.ProductionClaimedSignature.ToByteArray()),
+                typeof(ProductionCertificate)));
+
+        if (result.State != CommandStepState.Completed)
+            return ToCommandStatus(command, result);
+
+        result = await _stepProcessor.Process(
+            new CommandStep<ClaimCommand.Types.ClaimedEvent>(
+                command.Content.ConsumptionClaimedEvent.CertificateId.ToModel(),
+                new SignedEvent<ClaimCommand.Types.ClaimedEvent>(command.Content.ConsumptionClaimedEvent, command.Content.ConsumptionClaimedSignature.ToByteArray()),
+                typeof(ConsumptionCertificate)));
+
+        return ToCommandStatus(command, result);
```

Comment:
This step processor is a really cool approach - simple and understandable procedural process.